src/main/java/digital/slovensko/autogram/server/SignEndpoint.java

```diff
@@ -25,28 +25,26 @@ public SignEndpoint(Autogram autogram) {
 
     @Override
     public void handle(HttpExchange exchange) throws IOException {
-        exchange.getResponseHeaders().add("Access-Control-Allow-Origin", "*");
-        exchange.getResponseHeaders().add("Access-Control-Allow-Methods", "POST, OPTIONS");
-        exchange.getResponseHeaders().add("Access-Control-Allow-Headers", "Content-Type, Authorization");
-
-        // Allow preflight requests
-        if (exchange.getRequestMethod().equalsIgnoreCase("OPTIONS")) {
-            exchange.sendResponseHeaders(204, -1);
-            return;
-        }
-
         try {
             var body = EndpointUtils.loadFromJsonExchange(exchange, SignRequestBody.class);
+
             MimeType transformationOutputMimeTypeForXdc = null;
             if (body.getParameters().getContainer() != null) {
                 transformationOutputMimeTypeForXdc = DocumentVisualizationBuilder
                         .getTransformationOutputMimeType(body.getParameters().getTransformation());
             }
             var job = new SigningJob(body.getDocument(), body.getParameters(), new ServerResponder(exchange),
                     transformationOutputMimeTypeForXdc);
-            autogram.sign(job);
+            
+            if (body.getBatchId() != null) {
+                autogram.batchSign(job, body.getBatchId());
+            } else {
+                autogram.sign(job);
+            }
+
         } catch (JsonSyntaxException e) {
-            var response = ErrorResponse.buildFromException(new MalformedBodyException(e.getMessage(), e));
+            var response =
+                    ErrorResponse.buildFromException(new MalformedBodyException(e.getMessage(), e));
```

Comment:
Asi si vypni tych 80 znakov na riadok, lebo to rozbija taketo veci, co asi nikomu nevadia.

Reply:
uz som to nasiel, mal som nespravne prepisany nazov profilu 🤦